homeassistant/components/withings/sensor.py

```diff
@@ -369,9 +377,21 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the sensor config entry."""
-    coordinator: WithingsDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    measurement_coordinator: WithingsMeasurementDataUpdateCoordinator = hass.data[
+        DOMAIN
+    ][entry.entry_id][MEASUREMENT_COORDINATOR]
+
+    async_add_entities(
+        WithingsMeasurementSensor(measurement_coordinator, attribute)
+        for attribute in MEASUREMENT_SENSORS
+    )
+    sleep_coordinator: WithingsSleepDataUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ][SLEEP_COORDINATOR]
 
-    async_add_entities(WithingsSensor(coordinator, attribute) for attribute in SENSORS)
+    async_add_entities(
```

Comment:
Please collect all entities first and then make a single call to `async_add_entities`.